src/commands/githubActions/bumpUpstream/index.ts

```diff
@@ -55,180 +49,178 @@ export const gaBumpUpstream: CommandModule<
       type: "boolean"
     }
   },
-  handler: async (args): Promise<void> => await gaBumpUpstreamHandler(args)
+  handler: async (args): Promise<void> => gaBumpUpstreamHandler(args)
 };
 
 async function gaBumpUpstreamHandler({
   dir = defaultDir,
-  eth_provider,
-  use_fallback
+  eth_provider: userEthProvider,
+  use_fallback: useFallback,
 }: CliCommandOptions): Promise<void> {
-  // Check if buildSdkEnvFileName file exists
-  const templateArgs = readBuildSdkEnvFileNotThrow(dir);
 
-  const { manifest, format } = readManifest({ dir });
-  const compose = readCompose({ dir });
+  const { upstreamSettings, manifestData: { manifest, format }, compose, gitSettings, ethProvider } = await readInitialSetup({ dir, userEthProvider, useFallback });
+  printSettings(upstreamSettings, gitSettings, manifest, compose, ethProvider);
 
-  const upstreamRepos = templateArgs
-    ? [templateArgs._BUILD_UPSTREAM_REPO]
-    : parseCsv(manifest.upstreamRepo);
-  const upstreamArgs = templateArgs
-    ? [templateArgs._BUILD_UPSTREAM_VERSION]
-    : parseCsv(manifest.upstreamArg || "UPSTREAM_VERSION");
+  const upstreamRepoVersions = await getUpstreamRepoVersions(upstreamSettings);
+  if (isEmpty(upstreamRepoVersions)) {
+    console.log("There are no new versions to update");
+    return;
+  }
 
-  const githubActor = process.env.GITHUB_ACTOR || "bot";
-  const userName = githubActor;
-  const userEmail = `${userName}@users.noreply.github.com`;
+  const githubSettings = await getGitHubSettings(dir, upstreamRepoVersions);
+  const { branchName, repo } = githubSettings;
+  if (!(await isBranchNew({ branchName, repo }))) {
+    console.log(`Branch ${branchName} already exists`);
+    return;
+  }
 
-  const defaultEthProviders = ["remote", "infura"];
-  const ethProvider = eth_provider;
+  const composeVersionsToUpdate = updateComposeVersions(dir, compose, upstreamRepoVersions);
+  if (isEmpty(composeVersionsToUpdate)) {
+    console.log("All versions are up-to-date");
+    return;
+  }
 
-  const ethProviders = use_fallback
-    ? [ethProvider, ...defaultEthProviders.filter(p => p !== ethProvider)]
-    : [ethProvider];
+  updateManifest({ manifest, manifestFormat: format, composeVersionsToUpdate, dir, ethProvider });
 
-  console.log(`
-Arguments - ${JSON.stringify({
-    upstreamRepos,
-    upstreamArgs,
-    userName,
-    userEmail
-  })}
+  await prepareAndCommitChanges({
+    dir,
+    gitSettings,
+    composeVersionsToUpdate,
+    githubSettings,
+  });
 
-Manifest - ${JSON.stringify(manifest, null, 2)}
+  try {
+    await closeOldPrs(repo, branchName);
+  } catch (e) {
+    console.error("Error on closeOldPrs", e);
+  }
 
-Compose - ${JSON.stringify(compose, null, 2)}
-`);
+  const gitHead = await getGitHead();
```

Comment:
if you are not using `gitHead` nowehere else inside this loop why not calling it directly from `buildAndComment`

Reply:
You are right, done

Reply:
Because `buildAndComment` is also used in the build command and it is not always the git head